routes/cloudflareCache.js

```diff
@@ -0,0 +1,9 @@
+const express = require("express");
+const router = express.Router();
+const cloudflareCache = require("../controllers/cloudflareCache");
+const authenticate = require("../middlewares/authenticate");
+
+router.get("/", authenticate, cloudflareCache.fetchPurgedCacheMetadata);
+router.post("/", authenticate, cloudflareCache.purgeMembersCache);
```

Comment:
when are we planning to purge the cache, is it automatic, or when someone hits the endpoint? If it is based on hitting the endpoint shouldn't it be done by a superuser as any normal user can easily overload the endpoint? As per the Cloudflare API, we only get 30,000 purge requests in 24 hours.

Reply:
The whole idea behind this feature is, when someone is showing their page to some recruiter or someone else and if the page is cached of members site, then they can instantly purge the cache to see the updated page.
Also we have put the limit to user, in `24 hours` they are allowed to purge cache only 3 times.

Consider a scenario
- I have 3 requests available
- I have purged the cache 1 time now 2 more requests are remaining
- Now this 1 slot will be freed after 24 hours means from the current time after 24 hours I have again 3 remaining slots available to purge the cache.

Reply:
Got it.